### ISSUE TITLE
unable to boot into macOS after installing

### ISSUE THREAD

**Issue by tyrtok** [toleda/audio_CloverALC#128]
rig: fresh high sierra with apfs
mount efi, disable sip > installed > restart. 

verbose stuck at 
`installedGTK: GTK installed`
`setting BTCoex Config~~`

any reason  what i did wrong to not able to boot into os?

**Comment by just-some-tall-bloke** [comment]
Same issue here. Worked fine on 10.13 and 10.13.1, now hangs on boot on 10.13.2

**Comment by toleda** [comment]
@tyrtok
@just-some-tall-bloke

Nvidia web drivers installed (y/n)?
Lilu.kext installed (y/n)?

Software Update/Combo Update 10.13.2: after restart, rebuild kernel cache, restart. No need to run cloverALC; no AppleHDA.kext changes.
New install 10.13.2: have not tested yet.

**Comment by just-some-tall-bloke** [comment]
@toleda 

Nvidia web drivers installed (y/n)? y
Lilu.kext installed (y/n)? y

Latest versions of each. 

There may well be no changes to AppleHDA.kext, but my after installing the 10.13.2 combo I had no sound. Ran the cloverALC script, then the system didn't boot until I deleted AppleHDA.kext via recovery.

Telling me to reboot and then rebuild the kernel cache when I have a system that doesn't boot? I'll let you figure out the flaw with that plan ;)

**Comment by toleda** [comment]
cloverALC does not cause the boot failure. The kext cache built during the restart after instal does not boot.

10.13.2 Fixes:

**10.13.2 Desktop after install**:
1. Terminal/sudo touch /System/Library/Extensions && sudo kextcache -u /
2. Restart

**10.13.2 Boot fail after install**: 
1. Clover/Single User
2. return (root# appears)
3. /sbin/mount -uw /
4. touch /System/Library/Extensions
5. kextcache -u /
6. exit (boot continues to Desktop)
7.  Restart

**Comment by Yoshi2889** [comment]
Can confirm the boot after installing CloverHDA breaks. Following above instructions fixed it for me.

This is on a Gigabyte B150M-D3H with ALC892.

**Comment by upxdouglas121** [comment]
I can also confirm that the fix works with the Gigabyte H97M D3H (ALC 892) and i5 4460 with Intel HD 4600.

**Comment by usawa** [comment]
I've exaclty the same issue. In fact I've to install cloverALC and reboot to single at each boot, as the procedure doesn't seem to definitively fix the issue. Any idea ?

**Comment by toleda** [comment]
cloverALC rebuilds kext cache whenever a change is made that effects kernel cache.  The 10.13.2 Update is different from every previous update.  10.13.2 update boots with an installer (like a clean install). The installer rebuilds the kext cache without considering all kexts, specifically, non native kexts.  To avoid a failed boot, rebuild 10.13.2 kext cache and restart. Upon restart, cloverALC installs required files and rebuilds kext cache as expected. In my opinion, the 10.13.3 update is likely to use the previous method.

**Comment by steve3d** [comment]
I think 10.13.3 is different because the os will automatically rebuild the kernel cache.

Just try this, If you `sudo touch /System/Library/Extensions`, and after a few minutes,  run `sudo kextcache -u /` gives you nothing, 

but if you rebuild kernel cache right after touch the folder, then you will see there is some output.

and the install script is rebuild cache right after modified the kext. 

So I think that's why we need to manually rebuild again.

**Comment by toleda** [comment]
10.13.2 and 10.13.3 are the same and different than all previous updates. The difference is the install partition that contains the update, similar to a fresh install. That update partition does not contain the full OS. Unfortunately, without the full OS, any kext cache is invalid.

"if you rebuild kernel cache right after touch the folder"
Previously noted:
Terminal/sudo touch /System/Library/Extensions && sudo kextcache -u /

**Comment by sailor25462** [comment]
I was running into the same issue, after having updated to the latest 10.13.3.
After making a fresh install over the existing Setup, boot completed nicely. 
However, whenever I re-install the patch, even with re-building kept cache manually again, boot will fail afterwards.
Occasionally I get a kextd Stall for AppleAcpiCpu error and many lines of 
CoreStorageGroup::handleOpen messages.

I‘m running fusion drive!

Would be great to get some input as I’m running out of ideas...

**Comment by toleda** [comment]
"However, whenever I re-install the patch, even with re-building kext cache manually again, boot will fail afterwards."

You have described the method that causes the problem as described previously, full stop. 
Mandatory, restart before reinstalling patch.

**Comment by sailor25462** [comment]
> You have described the method that causes the problem as described previously, full stop.
> Mandatory, restart before reinstalling patch.

Thanks for the quick feedback!
I'll go through it again, then. Cross fingers.

Question: Any (easy) repair possible at all, when doing this kind of mistake? (rather than re-installing the whole system...)

**Comment by toleda** [comment]
See **"10.13.2 Boot fail after install"** above

**Comment by sumit52315** [comment]
Dear Toleda,

I have run "10.13.2 Boot fail after install" but not got login to desktop. Please help me.

Sumiy

**Comment by sumit52315** [comment]
Sorry Toleda, 

I am running os is "10.12.1" and run above command mentioned and not getting to login desktop.

Sumit

**Comment by toleda** [comment]
See "**10.13.2 Boot fail after install**" above

**Comment by sumit52315** [comment]
Dear Toleda, 

You mean I should run these commands:

10.13.2 Boot fail after install:

1. Clover/Single User
2. return (root# appears)
3. /sbin/mount -uw /
4. touch /System/Library/Extensions
5. kextcache -u /
6. exit (boot continues to Desktop)
7. Restart

I have run all 1-6 commands but did not get to desktop and also run command 7 by pressing hard restart button.

Toleda, I am running macOS Sierra 10.12.

Please help me.

Kind Regards,
Sumit

**Comment by toleda** [comment]
Command 7 is the result of 6; no action required.  
Because exit did not work, at least one of the previous commands did not execute properly or was entered incorrectly.  Nothing changed as the restart showed.
Try again.

**Comment by sumit52315** [comment]
Dear Toleda,

I have run again all the commands carefully and not getting to log in desktop.

1. Clover/Single User
2. return (root# appears)
3. /sbin/mount -uw /
4. touch /System/Library/Extensions
5. kextcache -u /
6. exit (boot continues to Desktop) **-- I have waited 15 minutes but not getting desktop login option just only showing mouse pointer.**
7. Restart **-- If not getting desktop then only option is to press hard button for shutdown or restart.**

Please suggest what should I do or I can send you complete video for the same.

Best Regards,
Sumit

**Comment by toleda** [comment]
Is there a # sign prompt when you enter exit?

Run the macOS installer over the current unbootable installation.

**Comment by sumit52315** [comment]
Dear Toleda,

No sign in prompt, only getting black screen with mouse pointer after exit command.

Thank you Toleda, I have placed order boot pen drive from [Amazon](https://www.amazon.in/gp/product/B079KFWWZ4/ref=oh_aui_detailpage_o00_s00?ie=UTF8&psc=1) and I will get it today by 9pm then get back you if need any help.

Thanks
Sumit

**Comment by toleda** [comment]
Answered a different question, The question; in root, is there a # sign prompt when you enter exit?

How did you install macOS without a pen drive?

**Comment by sumit52315** [comment]
Dear Toleda,

No, type exit command and press enter then get black screen only and nothing else.

**Comment by sumit52315** [comment]
I do not know how to install because my boss installed for me.

Now, I have bootable pendrive 10.13, bought from Amazon, but also not able to get fresh installation.

Please, let me know one thing, when I boot **Boot Recovery from Recovery HD** then keyboard and mouse do not work, I am using TVS Gold keyboard and Logitech B100 mouse.

**Comment by toleda** [comment]
No such item as a bootable pen drive 10.13, bought from Amazon.  Apple does not make a bootable pen drive. Return for a refund. Seller is not legitimate.

USB problem, try all USB ports.

**Comment by sumit52315** [comment]
When I bought it from Amazon, there is mentioned that it is not refundable, so how I can return it.

**Comment by toleda** [comment]
That was the clue, not legitimate.  Report to Amazon.

**Comment by sumit52315** [comment]
Okay, I will do it definitely.

Please, let me know one thing, when I boot from **Boot Recovery from Recovery HD** then keyboard and mouse do not work, I am using TVS Gold keyboard and Logitech B100 mouse.

I have enabled "Legacy USB Support" at Gygabyte Peripherals.

**Comment by toleda** [comment]
No idea.

**Comment by gameunite** [comment]
I tried installing using v0.4 but now It just boot like forever. Is it really the same with the 10.13.2? Cause I updated mine without any issue before. Only on this update 10.13.3 and with the 0.4 it started failing.

**Comment by toleda** [comment]
Yes. As of 1013.2, all Software Updates use an installer partition. 
Reboot after installer, mandatory,  before installing audio.
See **10.13.2+ Boot fail after install:** (above, post 5) for fix.
